source/FreeRTOS_ND.c

```diff
@@ -1270,4 +1270,97 @@
     }
 /*-----------------------------------------------------------*/
 
+    #if ( ipconfigREPLY_ICMP_PORT_UNREACHABLE != 0 )
+
+/**
+ * @brief Send an ICMPv6 destination unreachable message.
+ * @param[in] ucErrorCode: code field in message, fill in ipICMP_DEST_UNREACH_* from FreeRTOS_IPv6.h.
+ * @param[in] pxSourceNetworkBuffer: The network buffer carrying the packet which needs destination unreachable response.
+ *
+ * @return pdTRUE when a packets was successfully created.
+ */
+        /* MISRA Ref 8.9.1 [File scoped variables] */
+        /* More details at: https://github.com/FreeRTOS/FreeRTOS-Plus-TCP/blob/main/MISRA.md#rule-89 */
+        /* coverity[misra_c_2012_rule_8_9_violation] */
+        /* coverity[single_use] */
+        BaseType_t FreeRTOS_SendDestUnreachableIPv6( uint8_t ucErrorCode,
+                                                     NetworkBufferDescriptor_t * pxSourceNetworkBuffer )
+        {
+            NetworkBufferDescriptor_t * pxNetworkBuffer;
+            NetworkInterface_t * pxInterface;
+            ICMPPacket_IPv6_t * pxICMPPacket;
+            ICMPDestUnreachHeader_IPv6_t * pxICMPDestUnreachHeader_IPv6;
+            size_t uxPacketSize;
+            BaseType_t xResult = pdFAIL;
+
+            uxPacketSize = ipSIZE_OF_ETH_HEADER + ipSIZE_OF_IPv6_HEADER + sizeof( ICMPDestUnreachHeader_IPv6_t );
+
+            /* This is called from the context of the IP event task, so a block time
+             * must not be used. */
+            pxNetworkBuffer = pxGetNetworkBufferWithDescriptor( uxPacketSize, ndDONT_BLOCK );
+
+            if( pxNetworkBuffer != NULL )
+            {
+                /* The pointer points to source node's buffer. */
+                const IPPacket_IPv6_t * pxSourceIPHeader = NULL;
+
+                /* MISRA Ref 11.3.1 [Misaligned access] */
+                /* More details at: https://github.com/FreeRTOS/FreeRTOS-Plus-TCP/blob/main/MISRA.md#rule-113 */
+                /* coverity[misra_c_2012_rule_11_3_violation] */
+                pxSourceIPHeader = ( ( const IPPacket_IPv6_t * ) ( pxSourceNetworkBuffer->pucEthernetBuffer ) );
```

Comment:
~~~diff
-    pxSourceIPHeader = ( ( const IPPacket_IPv6_t * ) ( pxSourceNetworkBuffer->pucEthernetBuffer ) );
+    pxSourceIPHeader = ( const IPPacket_IPv6_t * ) pxSourceNetworkBuffer->pucEthernetBuffer;
~~~
I know that in many places in the IP-stack there are unnecessary brackets.